File: app/src/main/java/com/example/tetrisapp/ui/viewmodel/LobbyViewModel.java
package com.example.tetrisapp.ui.viewmodel;

import androidx.lifecycle.ViewModel;

import com.example.tetrisapp.model.local.model.UserInfo;

import java.util.ArrayList;
import java.util.List;

public class LobbyViewModel extends ViewModel {
    private final List<UserInfo> userList = new ArrayList<>();
    private String idToken = "";
    private String lobbyOwnerName;

    public void setIdToken(String idToken) {
        this.idToken = idToken;
    }

    public void setLobbyOwnerName(String lobbyOwnerName) {
        this.lobbyOwnerName = lobbyOwnerName;
    }

    public List<UserInfo> getUserList() {
        return userList;
    }

    public String getIdToken() {
        return idToken;
    }

    public String getLobbyOwnerName() {
        return lobbyOwnerName;
    }
}

File: app/src/main/java/com/example/tetrisapp/ui/viewmodel/LobbyViewModel.kt
package com.example.tetrisapp.ui.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.tetrisapp.model.local.model.UserInfo
import com.example.tetrisapp.util.FirebaseTokenUtil
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class LobbyViewModel: ViewModel() {
    private val mutableUsersLiveData: MutableLiveData<List<UserInfo>> by lazy {
        MutableLiveData(ArrayList())
    }
    val usersLiveData: LiveData<List<UserInfo>> = mutableUsersLiveData

    var token: String? = null
    var lobbyOwnerName: String? = null

    init {
        FirebaseTokenUtil.getFirebaseToken { token: String? ->
            this.token = token
        }
    }

    fun addUser(user: UserInfo) {
        viewModelScope.launch(Dispatchers.Main) {
            mutableUsersLiveData.value = mutableUsersLiveData.value?.toMutableList()?.let {
                it.add(user)
                it
            }
        }
    }

    fun removeUser(user: UserInfo) {
        viewModelScope.launch(Dispatchers.Main) {
            mutableUsersLiveData.value = mutableUsersLiveData.value?.toMutableList()?.let {
                it.remove(user)
                it
            }
        }
    }
}